### PR TITLE
samples: direct_test_mode: Implementation of the errata 117

### PR DESCRIPTION
**Pull request by KAGA164**
Implementation of the workaround for nRF5340 errata 117
in the Direct Test Mode sample.

### CODE REVIEW

**Comment by KAGA164** [comment]
@anangl Can we use `nrf_erratas.h' here and also definition for erratas like NRF53_ERRATA_117_ENABLE_WORKAROUND ?

**Comment by anangl** [comment]
> @anangl Can we use `nrf_erratas.h' here and also definition for erratas like NRF53_ERRATA_117_ENABLE_WORKAROUND ?

Yes, I see no reason why you should not.

**Review by anangl — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -169,6 +169,7 @@ Bluetooth samples
       * The front-end module test parameters are not set to their default value after the DTM reset command anymore.
       * Added the vendor-specific ``FEM_DEFAULT_PARAMS_SET`` command for restoring the default front-end module parameters.
       * Added possibility to build with the limited nRF21540 front-end module hardware pinout.
+      * Added workaround for nRF5340 revision 1 Errata 117.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:172]:
```suggestion
      * Added a workaround for nRF5340 revision 1 Errata 117.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/bluetooth/direct_test_mode/src/dtm.c [hunk 2]
```diff
@@ -1332,7 +1333,16 @@ static enum dtm_err_code phy_set(uint8_t phy)
...
-#endif
+#endif /* NRF52840_XXAA */
+
+#ifdef NRF53_ERRATA_117_ENABLE_WORKAROUND
+		/* Workaround for Errata 117 "RADIO: Changing MODE requires additional
+		 *  configuration" found at the Errata document for your device located at
+		 * https://infocenter.nordicsemi.com/index.jsp
+		 */
+		*((volatile uint32_t *)0x41008588) = *((volatile uint32_t *)0x01FF0080);
+#endif /* NRF53_ERRATA_117_ENABLE_WORKAROUND */
```
Comment (kapi-no) [inline samples/bluetooth/direct_test_mode/src/dtm.c:1344]: Consider defining a dedicated function for handling the errata activation (if necessary). The function can be called _errata_117_handle_. This way, we won't have as much boilerplate code and comment descriptions.

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (KAGA164) [reply doc/nrf/releases/release-notes-changelog.rst]: Done

File: samples/bluetooth/direct_test_mode/src/dtm.c (hunk 2 above)
Comment (KAGA164) [reply samples/bluetooth/direct_test_mode/src/dtm.c]: Good idea. I added a function for handling this errata

**Review by kapi-no — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Pull request merged**